### PR TITLE
HBASE-23102: Improper Usage of Map putIfAbsent

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 37s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 14s |  master passed  |
| +1 :green_heart: |  compile  |   1m 41s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 49s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 14s |  master passed  |
| +0 :ok: |  spotbugs  |   4m  1s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 18s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 16s |  The patch passed checkstyle in hbase-procedure  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  hbase-server: The patch generated 0 new + 11 unchanged - 4 fixed = 11 total (was 15)  |
| +1 :green_heart: |  checkstyle  |   0m 14s |  The patch passed checkstyle in hbase-rsgroup  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 33s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 38s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 16s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 45s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 38s |  hbase-procedure in the patch passed.  |
| -1 :x: |  unit  | 161m 51s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  unit  |   5m 52s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 236m 56s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-828/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/828 |
| JIRA Issue | HBASE-23102 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7e95d7c625b8 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-828/out/precommit/personality/provided.sh |
| git revision | master / 1d6f50fd86 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-828/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-828/1/testReport/ |
| Max. process+thread count | 4806 (vs. ulimit of 10000) |
| modules | C: hbase-procedure hbase-server hbase-rsgroup U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-828/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminServer.java [hunk 1]
```diff
@@ -575,32 +575,36 @@ public void removeServers(Set<Address> servers) throws IOException {
...
+        assignments.computeIfAbsent(currTable, key -> new HashMap<>())
+            .computeIfAbsent(currServer, key -> new ArrayList<>())
+            .add(currRegion);
       }
     }
 
     Map<ServerName, List<RegionInfo>> serverMap = Maps.newHashMap();
-    for(ServerName serverName: master.getServerManager().getOnlineServers().keySet()) {
-      if(rsGroupInfo.getServers().contains(serverName.getAddress())) {
+    for (ServerName serverName: master.getServerManager().getOnlineServers().keySet()) {
```
Comment (wchevreuil) [inline hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminServer.java]: These are more of style changes, not related to the actual jira issue. Let's address it in a separate issue, and keep this one focused on replacing _putIfAbsent_ by _computeIfAbsent_, where applicable.

**Comment by belugabehr** [comment]
@wchevreuil Thank you for your interest and review!

I have backed out the style changes.  Please review once more. Thanks!

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 38s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 52s |  master passed  |
| +1 :green_heart: |  compile  |   1m 49s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 58s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 50s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 16s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 15s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 34s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 40s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 40s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 14s |  The patch passed checkstyle in hbase-procedure  |
| +1 :green_heart: |  checkstyle  |   1m 18s |  hbase-server: The patch generated 0 new + 11 unchanged - 4 fixed = 11 total (was 15)  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  The patch passed checkstyle in hbase-rsgroup  |
| +1 :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 34s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 15s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 25s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 35s |  hbase-procedure in the patch passed.  |
| +1 :green_heart: |  unit  | 162m 55s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   6m 23s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 240m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-828/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/828 |
| JIRA Issue | HBASE-23102 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 48f885ef4671 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-828/out/precommit/personality/provided.sh |
| git revision | master / b292ffd14d |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-828/2/testReport/ |
| Max. process+thread count | 4867 (vs. ulimit of 10000) |
| modules | C: hbase-procedure hbase-server hbase-rsgroup U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-828/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**